src/mixins/widget.js

```diff
@@ -22,6 +22,11 @@ export const createWidgetMixin = ({ connector } = {}) => ({
       this.factory = connector(this.updateState, () => {});
       this.widget = this.factory(this.widgetParams);
       this.instantSearchInstance.addWidget(this.widget);
+
+      const { hydrated, started } = this.instantSearchInstance;
+      if ((!started && hydrated) || this.$isServer) {
```

Comment:
`hydrated` is never set back to `false`, it can leads to some issues since the InstantSearch instance is global and never reset to its initial state (an other potential solution is to reset the instance on `dispose`).

```
/search (server) -> SSR -> hydrate -> forceRender -> /about (client) -> /search -> forceRender
```

The last `forceRender` should not happen because the last results might be different from the one we want to render on a fresh page. We can have a blink with the stalled results from previous search and then the fresh results for the current page. Note that the blink only occurs when the request is not cached. When it's cached it looks like Vue is batching the two renders.

![blink](https://user-images.githubusercontent.com/6513513/51472434-c3848100-1d79-11e9-893a-49f683612b63.gif)

Reply:
TODO: clear helper on dispose